gateleen-core/src/main/java/org/swisspush/gateleen/core/util/StringUtils.java

```diff
@@ -233,18 +234,29 @@ private static String replacePlainStringWildcardConfigs(String contentWithWildca
      * "attribute" : 1234
      * </pre>
      *
-     * @param jsonWithWildcards the JSON String containing the attributes  with non string target type
+     * @param contentWithWildcards the JSON String containing the attributes  with non string target type
      *                          wildcards to replace
      * @param properties        the properties with the replacement values for the wildcards
      * @return the String with replaced wildcard values. Returns the input String when input String or
      * properties are <code>null</code>
      */
-    private static String replaceJsonNonStringAttributeWildcardConfigs(String jsonWithWildcards,
-        Map<String, Object> properties) {
+    private static String replaceJsonNonStringAttributeWildcardConfigs(String contentWithWildcards, Map<String, Object> properties) {
+        if (properties == null || contentWithWildcards == null) {
+            return contentWithWildcards;
+        }
+        // we don't touch it if not really necessary
+        if (contentWithWildcards.indexOf(JSON_NUMERIC_START_TOKEN) < 0) {
+            return contentWithWildcards;
+        }
         Engine engine = new Engine();
-        engine.setExprStartToken("\"$${");
-        engine.setExprEndToken("}\"");
-        return wildcardReplacementEngineIgnition(engine, jsonWithWildcards, properties);
+        engine.setExprStartToken(JSON_NUMERIC_START_TOKEN);
+        engine.setExprEndToken(JSON_NUMERIC_END_TOKEN);
+        String contentWithoutWildcards = wildcardReplacementEngineIgnition(engine, contentWithWildcards, properties);
+        // Note: due to the implicit string conversions of Java we just lost our escapes here and we must add
```

Comment:
(*trivial, SideNote*) Is this really "Java" doing this? Isn't it that 3rd-party "Engine" we're using?

Reply:
No, as far as I know, it is really the runtime doing this BS, even the compiler does something as well.
See here:  https://www.ibm.com/support/pages/when-use-double-backslash-escape-special-character-regular-expression

Overall a big mess.